### DISCUSSION TITLE
Questions on how to integrate Polish language (currently incubated)

### DISCUSSION THREAD

**Discussion by c0de-bender** [pretix/pretix#3711] [Q&A] [answered]
Hi, we started translating some strings to Polishon [translate.pretix.eu](https://translate.pretix.eu), because we want to integrate it. On the way we noticed that it's a lot easier for us to directly edit the files `src/pretix/locale/pl/LC_MESSAGES/django.po` and `src/pretix/locale/pl/LC_MESSAGES/djangojs.po`. I'm deploying from source, so we can use them for now, but of course we want to sync our translations back, so everyone can use them. I have a few questions about this:

1. Can we just edit these files and then ~~push them to~~ make a PR for the git repo? When this PR gets merged, will the changes in there be synced back to [translate.pretix.eu](https://translate.pretix.eu), or will the next push of [translate.pretix.eu](https://translate.pretix.eu) overwrite our changes? (Probably and hopefully not ^^)
2. Is editing translations in the `djangojs.po` file even necessary? [This](https://github.com/pretix/pretix/discussions/2946#discussioncomment-4473241) suggests it isn't, but on first glance it looks to me like this file does contain strings presented to end users.
3. Can you remove Polish from the list of incubated languages, once we are done? For now, I can just do this for us locally by editing the source code (or, better, [using the Pretix config file](https://docs.pretix.eu/en/latest/admin/config.html#translations)), but we'd like to make our translation available for others too. (I can also write an email to translate@pretix.eu once we are done.)

Thanks!

**Comment by raphaelm** [comment, marked as answer]
>  1. Can we just edit these files and then ~push them to~ make a PR for the git repo? When this PR gets merged, will the changes in there be synced back to [translate.pretix.eu](https://translate.pretix.eu), or will the next push of [translate.pretix.eu](https://translate.pretix.eu) overwrite our changes? (Probably and hopefully not ^^)

Yes, that *should* work, but it would be probably even safer to go through the "Upload" feature of translate.pretix.eu:
https://translate.pretix.eu/projects/pretix/pretix/pl/#upload

>  2. Is editing translations in the `djangojs.po` file even necessary?

Yes, should be translated as well.

>  3. Can you remove Polish from the list of incubated languages, once we are done?

Of course, we usually do that once a significant portion of the shop is translated. Feel free to email us so we don't overlook it.

**Reply by c0de-bender** [reply]
Great, thank you! :slightly_smiling_face:

Then we'll upload to the translation site and write you an email when we're done.

**Comment by c0de-bender** [comment]
@raphaelm Ah I forgot my most important question!

In the database I saw that the value of some string columns (e.g. in tables `pretixbase_organizer_settingsstore` and `pretixbase_event_settingsstore`) is a json looking like
```json
{"en": "", "de": "", "de-informal": "", "ar": "", "zh-hans": "", "zh-hant": "", "cs": "", "da": "", "nl": "", "nl-informal": "", "fr": "", "el": "", "id": "", "it": "", "lv": "", "nb-no": "", "pt-pt": "", "ro": "", "ru": "", "es": "", "tr": "", "uk": ""}
```
Looks like all non-incubated languages are in there.

Do we need a database migration script to add Polish in there, or will this be automatic?

Or are these strings only stored like this for backend settings and the user frontend texts are only stored with the selected languages anyway? In this case, because we only want end user translations, we wouldn't need this now.

**Reply by raphaelm** [reply]
This will be automatic

**Reply by c0de-bender** [reply]
Great, thanks!

**Reply by feldsam** [reply]
> This will be automatic

Hello, I added Slovank lang translations two days ago, it is in github repo already, but I can not select SK locale, because it is not here. What to do please?

**Reply by raphaelm** [reply]
I have now added Slovak to the list of selectable languages: https://github.com/pretix/pretix/commit/223f0956117815829f8fb481f68805a66c91bc2d

**Reply by feldsam** [reply]
Hi, thanks! When will this propagate to docker latest tag? I just downloaded latest one, but there is no Slovak in selectbox.

**Reply by raphaelm** [reply]
whenever docker hub finished building it
